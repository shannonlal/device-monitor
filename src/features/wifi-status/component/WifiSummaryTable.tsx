import React from 'react';
import { IonText, IonRow, IonCol, IonGrid } from '@ionic/react';
import { IWifiSummary } from '../../../interfaces/models';
import DeviceField from '../../../components/common/DeviceField';

type WifiSummaryTableProps = {
    ssidLabel: string;
    signalStength: string;
    wifiNetworks?: IWifiSummary[] | undefined;
};
const WifiSummaryTable: React.FC<WifiSummaryTableProps> = (props: WifiSummaryTableProps): React.ReactElement => {
    const getWifiSummary = (ssid: string, strength: number, id: number) => {
        const selectedWifi = (ssid: string) => {
            console.log('ssid called', ssid);
        };
        return (
            <DeviceField
                label={ssid}
                value={strength.toString()}
                key={`df-${id}-${ssid}-${strength}`}
                clickHandler={() => {
                    selectedWifi(ssid);
                }}
            />
        );
    };

    const getSummaryTable = (wifiNetworks: IWifiSummary[] | undefined) => {
        if (typeof wifiNetworks === 'undefined') {
            return;
        }

        const summaryNetworks = wifiNetworks.map((wifi, i) => {
            return getWifiSummary(wifi.ssid, wifi.signalStrength, i);
        });

        return summaryNetworks;
    };
    return (
        <IonGrid>
            <IonRow>
                <IonCol>
                    <IonText>{props.ssidLabel}</IonText>
                </IonCol>
                <IonCol>
                    <IonText>{props.signalStength}</IonText>
                </IonCol>
            </IonRow>
            {getSummaryTable(props.wifiNetworks)}
        </IonGrid>
    );
};

export default WifiSummaryTable;
