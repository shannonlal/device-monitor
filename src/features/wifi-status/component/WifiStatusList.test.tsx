// import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import WifiSummaryTable from './WifiSummaryTable';
// import { IWifiSummary } from '../../../interfaces/models';
// import { getWifiBridge } from '../../../drivers/DriverFactory';

/*const fakeWifiBridge = (): Promise<IWifiSummary[]> => {
    console.log('Calling Fake Wifi Bridge');
    const mockWifSummary: IWifiSummary[] = [
        {
            ssid: 'Test Wifi 4',
            signalStrength: 1,
        },
        {
            ssid: 'Test Wifi 5',
            signalStrength: 2,
        },
    ];
    return Promise.resolve(mockWifSummary);
};

jest.mock('../../../drivers/DriverFactory', () => ({
    ...jest.requireActual('../../../drivers/DriverFactory'),
    getWifiBridge: jest.fn(),
}));

const getWifiBridgeMock = getWifiBridge as jest.Mock;
*/
//const getWifiBridgeMock = getWifiBridge as jest.Mock;

// getWifiBridgeMock.mockImplementation(() => fakeWifiBridge);

//beforeAll(() => {
//    getWifiBridge.mockImplementation(fakeWifiBridge);
//});
describe('WifiStatusList', () => {
    test('rendering a empty status list', async () => {
        const ssidLabel = 'Network Name';
        const signalStrengthLabel = 'Signal Strength';

        const { asFragment } = render(<WifiSummaryTable ssidLabel={ssidLabel} signalStength={signalStrengthLabel} />);

        expect(asFragment()).toMatchSnapshot();
    });

    /*test('rendering a wifi status list', async () => {
        const ssidLabel = 'Network Name';
        const signalStrengthLabel = 'Signal Strength';

        const { asFragment } = render(<WifiSummaryTable ssidLabel={ssidLabel} signalStength={signalStrengthLabel} />);

        await waitFor(() => screen.findByText('Test Wifi 4'));
        //expect(await screen.findByText('Test Wifi 4')).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });*/
});
