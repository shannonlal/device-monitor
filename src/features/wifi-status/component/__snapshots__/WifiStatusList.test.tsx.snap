// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WifiStatusList rendering a empty status list 1`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Network Name
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Signal Strength
        </ion-text>
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;

exports[`WifiStatusList rendering a wifi status list 1`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Network Name
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Signal Strength
        </ion-text>
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;
